src/Components/test/E2ETest/Tests/SectionsTest.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.E2ETesting;
+using Xunit.Abstractions;
+using Microsoft.AspNetCore.Components.E2ETest;
+using OpenQA.Selenium;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.Tests;
+public class SectionsTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
```

Comment:
I can think of a few additional scenarios that might be worth testing:
* Multiple `SectionContent` instances render to the same `SectionOutlet`:
  * If `IsDefaultContent` is `false` for a `SectionContent`, it will override existing content in its corresponding `SectionOutlet`.
  * If `DefaultContent` is `true`, it will not replace existing content in the `SectionOutlet`, but its content _will_ get rendered if all previously-registered `SectionContent`s (including those with `DefaultContent == true`) are disposed.
* The `Name` parameter of a `SectionContent` instance gets changed, causing it to match a different `SectionOutlet`.
* The `IsDefaultContent` parameter of a `SectionContent` gets changed (I actually don't think we handle this case currently... should we?)
* A `SectionContent` attempts to render to a `SectionOutlet` that doesn't exist.
  * The `SectionOutlet` never existed.
  * The `SectionOutlet` gets disposed before the `SectionContent` does.
* All `SectionContent` instances rendering to a `SectionOutlet` get disposed.

Reply:
>The IsDefaultContent parameter of a SectionContent gets changed (I actually don't think we handle this case currently)

Yes, we currently don't handle this case. I'm working on this.

I covered all these cases.

Could you please clarify what is expected to happen in case when two `SectionContent` instances with the same `Name` are set as default (`IsDefaultContent=true`) ?

Reply:
> Could you please clarify what is expected to happen in case when two `SectionContent` instances with the same `Name` are set as default (`IsDefaultContent=true`) ?

The way the current implementation was designed to work, the earliest-initialized `SectionContent` with `IsDefaultContent=true` should be rendered with higher priority. Or, in other words, the last-initialized default `SectionContent` is "more default" than the previous default `SectionContent`, in the sense that it's treated as the default content when the previous default `SectionContent` gets disposed.